### ISSUE TITLE
[FEATURE REQUEST] Hovering over an icon on the panel/taskbar with a file selected, automatically raises/brings to focus that minimized/not on focus app

### ISSUE THREAD

**Issue by Milli189** [home-sweet-gnome/dash-to-panel#2218]
**Describe the idea or feature you'd like**
If I want to drag and drop a file into an app that is minimized or not on focus anymore, I can:
1 select the file from the Files app
2 drag it on top the icon of the minimized app I want to import it to
3 show me the windows preview so I can hover over the one that I want and it will automatically raise it or put it to focus
4 and import the file into the app

currently it's not happening, as you can see in this short video I've just recorded:

https://github.com/user-attachments/assets/8d64173f-42fa-45ab-945d-46b75dd7190a

**Is your feature request related to a problem? Please describe.**
the current problem is that if I want to import a file to a minimized app I have to:
1 stop dragging the file I was dragging, (because of muscle memory coming from Windows)
2 click on the icon of the desired app to put it on focus, 
3 click on the icon of files to put that on focus on top of the desired app, 
4 and then drag the file again on top of the app,

and this takes more effort to do than just dragging the file on the icon of the app and make it maximize/focus automatically for me

**Additional context**
this feature would really be super convenient for me

**Comment by charlesg99** [comment]
Yes this feature used to work fine everywhere, then one day it broke on wayland, then later it broke on X11 and started to work again on wayland. Call it gnome-shell "improvements". There are a lot of issues opened on gnome repositories about drag and drop, most of them ignored. All I'm saying is that until mutter's dnd works correctly on both x11 and wayland, I don't see the point of fixing this to see it break again with the next gnome-shell version. I wish that wasn't the case though cause I like this feature.

**Comment by charlesg99** [comment]
Well, I restored showing the overview when dragging a file over the panel and added that a single window of an appicon gets focused on hover. It only works on wayland :( 

Can you please try it on the master branch and tell me if that works for you?

**Comment by Milli189** [comment]
> Can you please try it on the master branch and tell me if that works for you?

it does work but the overview is very annoying, I just need the window to raise that's it

**Comment by charlesg99** [comment]
Makes sense, added to master. The only way the overview could show is if your icon has more than one window attached to it. In this case the overview shows up so you can choose on which of the icon's windows to drop your file.

**Comment by Milli189** [comment]
> Makes sense, added to master. The only way the overview could show is if your icon has more than one window attached to it. In this case the overview shows up so you can choose on which of the icon's windows to drop your file.

yeah that makes sense too.

I've tried it and it works well, good job!
